### PR TITLE
Setup JWT-based authentication

### PR DESCRIPTION
**Pull request by abaniuszewicz**
[Nick Chapsas (YT): Setting up JWT support (Authentication)](https://www.youtube.com/watch?v=M6AkbBaDGJE&list=PLUOequmGnXxOgmSDWU7Tl6iQTsOtyjtwU&index=10)
[Nick Chapsas (YT): User Registration and Controller Auth](https://www.youtube.com/watch?v=ARvsBUBioT0&list=PLUOequmGnXxOgmSDWU7Tl6iQTsOtyjtwU&index=11)
[ASP.NET Core Identity Without Entity Framework](https://markjohnson.io/articles/asp-net-core-identity-without-entity-framework/)
[Overview of Custom Storage Providers for ASP.NET Identity](https://docs.microsoft.com/en-us/aspnet/identity/overview/extensibility/overview-of-custom-storage-providers-for-aspnet-identity)

### CODE REVIEW

File: backend/DevUp/DevUp.Infrastructure.Postgres/JwtIdentity/Validators/UsernameValidator.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using DevUp.Infrastructure.Postgres.JwtIdentity.Dtos;
+using Microsoft.AspNetCore.Identity;
+
+namespace DevUp.Infrastructure.Postgres.JwtIdentity.Validators
+{
+    internal class UsernameValidator : IUserValidator<UserDto>
+    {
+        public async Task<IdentityResult> ValidateAsync(UserManager<UserDto> manager, UserDto user)
```
Comment (Goush) [inline backend/DevUp/DevUp.Infrastructure.Postgres/JwtIdentity/Validators/UsernameValidator.cs:8]: Could be replaced with existing solution => https://docs.fluentvalidation.net/

**Review by Goush — CHANGES_REQUESTED** [review]

File: backend/DevUp/DevUp.Api/V1/Controllers/Identity/IdentityController.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+using DevUp.Domain.Identity;
+using DevUp.Domain.Identity.Exceptions;
+using DevUp.Infrastructure.Postgres.JwtIdentity.Results;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DevUp.Api.V1.Controllers.Identity
+{
+    [Route("api/v1/[controller]")]
+    [ApiController]
+    public class IdentityController : ControllerBase
```
[the remark this answers is not in the document]
Comment (abaniuszewicz) [reply backend/DevUp/DevUp.Api/V1/Controllers/Identity/IdentityController.cs:14]: We'll stick to MWE implementation for now. We'll add `.Refresh()` later on.

File: backend/DevUp/DevUp.Infrastructure/Documentation/SwaggerInstaller.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OpenApi.Models;
+
+namespace DevUp.Infrastructure.Documentation
+{
+    public static class SwaggerInstaller
```
[the remark this answers is not in the document]
Comment (abaniuszewicz) [reply backend/DevUp/DevUp.Infrastructure/Documentation/SwaggerInstaller.cs:7]: Make sure you add `Bearer <token>` when you authenticate.

File: backend/DevUp/DevUp.Infrastructure.Postgres/JwtIdentity/Validators/UsernameValidator.cs (hunk 1 above)
Comment (abaniuszewicz) [reply backend/DevUp/DevUp.Infrastructure.Postgres/JwtIdentity/Validators/UsernameValidator.cs:8]: Good suggestion. Will be handled as part of #28. Let's ignore it on this one for now.

**Pull request merged**